cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/SessionConfiguration.java

```diff
@@ -71,82 +73,105 @@ public class SessionConfiguration {
 
     @Value("${dmp.tokens.create_gml_session}")
     private String gmlCreateSession;
-    
+
+    @Value("${dmp.create_session_initial_response_timeout}")
+    private Integer dmpCreateSessionInitialResponseTimeout;
+
+    @Value("${dmp.create_session_maximum_response_timeout}")
+    private Integer dmpCreateSessionMaximumResponseTimeout;
+
+    @Value("#{jobParameters[dropDeadInstantString]}")
+    private String dropDeadInstantString;
+
     @Value("${dmp.tokens.retrieve_variants.impact}")
     private String retrieveVariantsImpact;
-    
+
     @Value("${dmp.tokens.retrieve_variants.rdts}")
     private String retrieveVariantsRaindance;
-    
+
     @Value("${dmp.tokens.retrieve_variants.heme}")
     private String retrieveVariantsHeme;
-    
+
     @Value("${dmp.tokens.retrieve_variants.archer}")
     private String retrieveVariantsArcher;
 
     @Value("${dmp.tokens.retrieve_variants.access}")
     private String retrieveVariantsAccess;
-    
+
     @Value("${dmp.tokens.retrieve_master_list.impact}")
     private String dmpMasterListImpact;
-    
+
     @Value("${dmp.tokens.retrieve_master_list.rdts}")
     private String dmpMasterListRaindance;
-    
+
     @Value("${dmp.tokens.retrieve_master_list.heme}")
     private String dmpMasterListHeme;
-    
+
     @Value("${dmp.tokens.retrieve_master_list.archer}")
     private String dmpMasterListArcher;
 
     @Value("${dmp.tokens.retrieve_master_list.access}")
     private String dmpMasterListAccess;
 
-    Logger log = Logger.getLogger(SessionConfiguration.class);
+    private Logger log = Logger.getLogger(SessionConfiguration.class);
 
     public final static String SESSION_ID = "cvrSessionId";
     public final static String GML_SESSION = "gmlSessionId";
 
+    private void logCreateSessionFailure(String dmpServerName, int numberOfRequestsAttempted, String message) {
+        log.error(String.format("Error creating CVR session for server %s (after %d attempts) : %s", dmpServerName, numberOfRequestsAttempted, message));
+    }
+
     /*
     * FULL URL: server_name/create_session/user_name/password/TYPE
     * The TYPE can be 0 or 1, 0 is de-identified, 1 is identified (for clinical information)
     * for CVR fetch for the portal, use 0
     */
-    private String dmpUrl;
-    private String gmlUrl;
 
-    // Gets the sessionId from CVR
+    /* Gets the sessionId from CVR
+     * TODO: this approach creates the session during the Bean Initialization phase (before actual application startup). It also
+     *       constrains each run of the cvr pipeline to having only a single session. Sessions expire after 3 hours, and this
+     *       does not allow for the possibility of begining subsequent sessions for later exchanges with the cvr servers if the
```

Comment:
This means that we don't automatically renew a session and all work needs to be complete within 3 hrs.  Typo "beginning"

Reply:
Correct - that is the current behavior. One (single) session is obtained, and it expires in 3 hours (according to the comment in the response from lynx). Our code only attempts to obtain a session during the bean wiring phase of application startup. Improving this behavior so that session expiration does not prevent continued progress in a single run could be a future enhancement.

Typo corrected.